File: Numbers/fibonacci.c
#include <stdio.h>

int fibonacci(int n);

void fibonaccis(int n);

int main(void) {
	int n;
	printf("How many numbers do you need? ");
	scanf("%d", &n);
	fibonaccis(n);
	return 0;
}

int fibonacci(int n) {
	//http://en.wikipedia.org/wiki/Fibonacci_number
	// The first two number choiced be 0 and 1 
	if (0 == n) {
		return 0;
	} else if (1 == n) {
		return 1;
	} else {
		return fibonacci(n-1) + fibonacci(n-2);
	}
}

void fibonaccis(int n)
{
	int i = 0;
	while(i < n) {
	 	printf("%d\t", fibonacci(i++));
	}
	printf("\n");
}

File: Numbers/fibonacci.go
/*
* Fibonacci Sequence - Enter a number and have the
* program generate the Fibonacci sequence to that number
* or to the Nth number
 */

package main

import (
	"fmt"
)

func fibonacci(n int) int {
	if n == 0 {
		return 1
	} else if n == 1 {
		return 1
	} else {
		return fibonacci(n-1) + fibonacci(n-2)
	}
}

func main() {
	fmt.Println("How many numbers do you need? ")
	n := 0
	fmt.Scanln(&n)
	numbers := []int{}
	for i := 0; i < n; i++ {
		numbers = append(numbers, fibonacci(i))
	}
	fmt.Println(numbers)
}
